crates/wysiwyg/src/dom/unicode_string.rs

```diff
@@ -59,6 +59,42 @@ pub trait UnicodeStr:
 
     // Should really be `-> Self::Chars<'a>`, but that requires GATs
     fn chars(&self) -> Box<dyn Iterator<Item = char> + '_>;
+
+    /// Returns the length of the char in indices of the current encoding
+    fn char_len(&self, char: &char) -> usize;
+
+    /// Splits the current UnicodeStr into graphemes (visual characters), from start to end indexes
+    fn graphemes(&self, start: usize, end: usize) -> Vec<&Self> {
+        let mut ret = Vec::new();
+        let mut start = start;
+        let mut offset = 0;
+        let mut chars = self.chars().peekable();
+        while chars.peek().is_some() {
+            let c = chars.next().unwrap();
```

Comment:
Can we not do `for char in chars` ?

Reply:
I tried doing that at first, but we need to know both the current and the next character (checking for ZWJ) to understand if the current slice should be added as a grapheme or not. The easiest way I found to do that was using `peek()`.

Reply:
Ah, I missed that you were using both, sorry.